src/Algebra/Solver/Ring/Simple/Reflection.agda

```diff
@@ -0,0 +1,223 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Solver.Ring.Simple.Reflection where
+
+open import Agda.Builtin.Reflection
+open import Reflection.Helpers
+
+open import Algebra.Solver.Ring.Simple.Solver renaming (solve to solve-fn)
+open AlmostCommutativeRing
+
+open import Data.NatSet
+
+open import Data.Fin   as Fin   using (Fin)
+open import Data.Vec   as Vec   using (Vec; _∷_; [])
+open import Data.List  as List  using (List; _∷_; [])
+open import Data.Maybe as Maybe using (Maybe; just; nothing; fromMaybe)
+open import Agda.Builtin.Nat    using (_<_)
+open import Data.Nat            using (ℕ; suc; zero)
+open import Data.Bool           using (Bool; if_then_else_; true; false)
+open import Data.Unit           using (⊤)
+open import Data.String         using (String)
+open import Data.Product        using (_,_)
+open import Function
+
+module Internal where
```

Comment:
The internal module hierarchy in this module is **very** confusing. I think it would be better to reduce the amount of nesting going on here. Also some additional comments wouldn't go amiss. For instance what does `Ring⇓` do?